src/arrayMethodSort.js

```diff
@@ -5,7 +5,26 @@
  */
 function applyCustomSort() {
   [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+    for (let i = 0; i < this.length - 1; i++) {
+      for (let j = 0; j < this.length - i - 1; j++) {
+        let compareResult;
+
+        if (compareFunction !== undefined) {
+          compareResult = compareFunction(this[j], this[j + 1]);
+        } else {
+          compareResult = (String(this[j]) > String(this[j + 1]));
+        }
```

Comment:
set for defautlt value for  compareFunciton. it compareFunction will not be passed to sort method - will work default value. 
```
[].__proto__.sort2 = function(compareFunction = (a, b) => String(a) > String(b)) {
```
...
```suggestion
          compareResult = compareFunction(this[j], this[j + 1]);
```

Reply:
with default argument you don't need anymore to chekc on undefined compare function. compareFunciton never will be undefined.
```
const compareResult = compareFunction(this[j], this[j + 1]);
```